From: Termux User <termux@termux.dev>
Date: Wed, 29 Jul 2026 21:30:00 +0000
Subject: [PATCH 2/2] Add Android/bionic compatibility to CMakeLists.txt

Add an Android (Termux) system block following the existing pattern
for distro-specific configurations (FEDORA, DEBIAN, SUSE, etc.).

This block defines __ANDROID__ which is needed for bionic libc
compatibility - certain features like stdio cookie functions are
not available on Android and need to be conditionally disabled.

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1111111..2222222 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,6 +133,12 @@ IF (HAIKU)
 ADD_DEFINITIONS (-DHAIKU)
 ENDIF (HAIKU)
 
+IF (CMAKE_SYSTEM_NAME STREQUAL "Android")
+    MESSAGE (STATUS "Building for Android/Termux")
+    ADD_DEFINITIONS (-D__ANDROID__)
+    SET (ENABLE_RPMDB ON)
+ENDIF (CMAKE_SYSTEM_NAME STREQUAL "Android")
+
 SET (CMAKE_MACOSX_RPATH ON)
 
 IF (${have_system} STREQUAL x)
-- 
2.47.0
